### PR TITLE
src: refactor builtins

### PR DESCRIPTION
**Pull request by devsnek**
- Refactor builtins to use a more organized registration system
- Make sure overridden console methods still call the original methods
- Remove duplicate library definitions in main.rs

Future work will need to be done to separate out the cli parts from the core parts and things of that nature.

### CODE REVIEW

**Comment by MylesBorins** [comment]
Opened an issue about a broader improvement to docs

https://github.com/jstime/jstime/issues/27

File: src/script.rs [hunk 1]
```diff
@@ -7,26 +7,23 @@ use crate::binding;
...
-        let msg = v8::Exception::create_message(tc_scope, exception);
-        let error_message = msg.get(tc_scope).to_rust_string_lossy(tc_scope);
-        eprintln!("{}", &error_message);
-        return "".to_string();
-    }
+    v8::Script::compile(scope, code, Some(&origin)).and_then(|script| script.run(scope))
+}
 
-    let script = script.unwrap();
+pub fn run_js_in_scope(scope: &mut v8::HandleScope, js: &str, filepath: &str) -> String {
```
Comment (MylesBorins) [inline src/script.rs:23]: this public api requires a HandleScope to run, I don't believe we have API that currently returns one... or am I mistaken.

**Comment by devsnek** [comment]
I mean the additional api surface shouldn't really be public, it's just a side effect of the current structure, and I'm planning to fix it.

**Comment by MylesBorins** [comment]
Ok cool. then let's land this and refactor. 

I do like the idea of having a public API that allows folks to do things such as set flags on V8 and boostrap specific built ins.

**Pull request merged**